frontend/middleware_validatesubscription.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/Azure/ARO-HCP/internal/api"
+	"github.com/Azure/ARO-HCP/internal/api/arm"
+	"github.com/Azure/ARO-HCP/internal/api/subscription"
+)
+
+type SubscriptionStateMuxValidator struct {
+	cache *Cache
+}
+
+func NewSubscriptionStateMuxValidator(c *Cache) *SubscriptionStateMuxValidator {
+	return &SubscriptionStateMuxValidator{
+		cache: c,
+	}
+}
+
+func (s *SubscriptionStateMuxValidator) MiddlewareValidateSubscriptionState(w http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+	subscriptionId := r.PathValue(strings.ToLower(PathSegmentSubscriptionID))
```

Comment:
Is lowercasing the path segment name going break the lookup?

Reply:
This middleware will be done after the `lowercase` middleware, since it is in the `postMux` flow.

Reply:
Should it be `strings.ToLower(r.PathValue(PathSegmentSubscriptionID))` then?  The path value key isn't lowercased, only the value.

Reply:
No. At least from what I see in the debugger, all of the keys for PathValue (r.pat.segments) are lowercased, as well as their matched value.